tests/handlers/test_password_providers.py

```diff
@@ -817,6 +824,35 @@ def test_3pid_allowed(self):
         self._test_3pid_allowed("rin", False)
         self._test_3pid_allowed("kitay", True)
 
+    def test_display_name(self):
+        """Tests that the get_display_name_for_registration callback can define the
+        display name of a user when registering.
+        """
+        self._setup_get_name_for_registration(
+            callback_name=self.CALLBACK_DISPLAYNAME,
+        )
```

Comment:
Should we be using the return value of the mock (e.g. ensuring it is called or something?)